src/io.c

```diff
@@ -24,6 +24,10 @@
 #define DISPATCH_IO_DEBUG DISPATCH_DEBUG
 #endif
 
+#ifndef PAGE_SIZE
+#define PAGE_SIZE 4096
```

Comment:
Very interesting.  PAGE_SIZE is not defined in sys/user.h on arm.  It's referenced in a few places, but I can't see how it's defined in those cases.

```
 wdillon@tegra-ubuntu  /usr/include  ack-grep "PAGE_SIZE"
scsi/sg.h
188:#define SG_SCATTER_SZ (8 * 4096)  /* PAGE_SIZE not available to user */
190:   The value must be a power of 2 and <= (PAGE_SIZE * 32) [131072 bytes on
191:   i386]. The minimum value is PAGE_SIZE. If scatter-gather not supported
194:   PAGE_SIZE by calling getpagesize() defined in unistd.h . */

elf.h
1882:#define PF_HP_PAGE_SIZE   		0x00100000

arm-linux-gnueabihf/bits/xopen_lim.h
50:   PAGE_SIZE	Size of bytes of a page.

arm-linux-gnueabihf/bits/confname.h
135:#define    	_SC_PAGE_SIZE  			_SC_PAGESIZE

asm-generic/shmparam.h
4:#define SHMLBA PAGE_SIZE     	 /* attach addr a multiple of this */

linux/resource.h
71:#define MLOCK_LIMIT 	((PAGE_SIZE > 64*1024) ? PAGE_SIZE : 64*1024)

linux/vhost.h
58:#define VHOST_PAGE_SIZE 0x1000

linux/kvm.h
352:   	((PAGE_SIZE - sizeof(struct kvm_coalesced_mmio_ring)) / \
499:#define KVM_PPC_PAGE_SIZES_MAX_SZ  	8
509:   	struct kvm_ppc_one_page_size enc[KVM_PPC_PAGE_SIZES_MAX_SZ];
512:#define KVM_PPC_PAGE_SIZES_REAL    		0x00000001
519:   	struct kvm_ppc_one_seg_page_size sps[KVM_PPC_PAGE_SIZES_MAX_SZ];

linux/binfmts.h
14:#define MAX_ARG_STRLEN (PAGE_SIZE * 32)
```